applications/tari_base_node/src/grpc/base_node_grpc_server.rs

```diff
@@ -995,32 +995,25 @@ impl tari_rpc::base_node_server::BaseNode for BaseNodeGrpcServer {
     ) -> Result<Response<tari_rpc::SyncInfoResponse>, Status> {
         debug!(target: LOG_TARGET, "Incoming GRPC request for BN sync data");
 
-        let mut channel = self.state_machine_handle.get_status_info_watch();
-
-        let mut sync_info: Option<BlockSyncInfo> = None;
-
-        if let Some(info) = channel.recv().await {
-            sync_info = info.state_info.get_block_sync_info();
-        }
-
-        let mut response = tari_rpc::SyncInfoResponse {
-            tip_height: 0,
-            local_height: 0,
-            peer_node_id: vec![],
-        };
-
-        if let Some(info) = sync_info {
-            let node_ids = info
-                .sync_peers
-                .iter()
-                .map(|x| x.to_string().as_bytes().to_vec())
-                .collect();
-            response = tari_rpc::SyncInfoResponse {
-                tip_height: info.tip_height,
-                local_height: info.local_height,
-                peer_node_id: node_ids,
-            };
-        }
+        let response = self
```

Comment:
Bug here - the call to `channel.recv().await` would wait for a channel update. - the state machine would typically do this if in listening state every few seconds (chain metadata service drives this). We just want to return whatever is the current state from this GRPC call.